services/src/tasks/mod.rs

```diff
@@ -0,0 +1,141 @@
+mod error;
+mod in_memory;
+
+use crate::error::Result;
+pub use error::TaskError;
+use geoengine_datatypes::{error::ErrorSource, identifier};
+pub use in_memory::{InMemoryTaskDb, InMemoryTaskDbContext};
+use serde::{Deserialize, Serialize, Serializer};
+use std::{fmt, sync::Arc};
+
+/// A database that allows registering and retrieving tasks.
+#[async_trait::async_trait]
+pub trait TaskDb<C: TaskContext>: Send + Sync {
+    async fn register(&self, task: Box<dyn Task<C>>) -> Result<TaskId, TaskError>;
```

Comment:
As this method also starts the task I think it should reflect that in the name

Reply:
[cfa6775](https://github.com/geo-engine/geoengine/pull/555/commits/cfa6775646267d1e5f1726fc3173297c827a192f)

Reply:
is "schedule" really better? I mean the tasks are not scheduled for some future execution, but they are started right then, or not? Why not call it run_task or something like that?

Reply:
Because in the future, we would also like to postpone running things if the server is packed.
Currently, we just run it directly.

Reply:
but this is not apparent from the function name. Maybe it's not important, but I thought it was.

How would it be decided when to run the task anyhow? Will there be priorities or an additional parameter (e.g. enum WHEN { now, time {}, whenever }? Maybe we could introduce that already and allow only the now variant? Maybe also yagni though.